src/game_state/AxisAlignedBox.cpp

```diff
@@ -3,30 +3,51 @@
 #include "game_state/AxisAlignedBox.h"
 
 #include <algorithm>
+#include <glm/ext.hpp>
+
+// static
+bool AxisAlignedBox::IsColliding(const AxisAlignedBox& one,
+                                 const AxisAlignedBox& two) {
+  return one.min.x <= two.max.x && one.max.x >= two.min.x &&
+         one.min.y <= two.max.y && one.max.y >= two.min.y &&
+         one.min.z <= two.max.z && one.max.z >= two.min.z;
+}
 
 AxisAlignedBox::AxisAlignedBox(std::shared_ptr<Shape> model,
                                glm::mat4 transform) {
-  min_x = min_y = min_z = FLT_MAX;
-  max_x = max_y = max_z = FLT_MIN;
-  const std::vector<unsigned>& pos_buf = model->GetElements();
+  bool first = true;
+  std::vector<float> pos_buf = model->GetPositions();
   for (int i = 0; i < pos_buf.size(); i += 3) {
     glm::vec4 position(pos_buf[i], pos_buf[i + 1], pos_buf[i + 2], 1.0f);
+    // TODO(jarhar): should we consider what happens to the 4th coordinate
+    // after the transform?
     glm::vec4 new_position = transform * position;
-    max_x = std::max(max_x, new_position.x);
-    min_x = std::min(min_x, new_position.x);
-    max_y = std::max(max_y, new_position.y);
-    min_y = std::min(min_y, new_position.y);
-    max_z = std::max(max_z, new_position.z);
-    min_z = std::min(min_z, new_position.z);
+    if (first) {
+      first = false;
+      min.x = max.x = new_position.x;
+      min.y = max.y = new_position.y;
+      min.z = max.z = new_position.z;
+    } else {
+      max.x = std::max(max.x, new_position.x);
+      min.x = std::min(min.x, new_position.x);
+      max.y = std::max(max.y, new_position.y);
+      min.y = std::min(min.y, new_position.y);
+      max.z = std::max(max.z, new_position.z);
+      min.z = std::min(min.z, new_position.z);
+    }
   }
 }
 
 AxisAlignedBox::~AxisAlignedBox() {}
 
-// static
-bool AxisAlignedBox::IsColliding(const AxisAlignedBox& one,
-                                 const AxisAlignedBox& two) {
-  return (one.min_x <= two.max_x && one.max_x >= two.min_x) &&
-         (one.min_y <= two.max_y && one.max_y >= two.min_y) &&
-         (one.min_z <= two.max_z && one.max_z >= two.min_z);
+glm::vec3 AxisAlignedBox::GetMin() {
+  return min;
+}
+
+glm::vec3 AxisAlignedBox::GetMax() {
+  return max;
+}
+
+std::string AxisAlignedBox::ToString() {
```

Comment:
lol is the java. I've got some debug render code for this though so you can actually visualize the bounding boxes I'll add that later